tests/gorepo/gorepo_test.go

```diff
@@ -24,7 +24,7 @@ func TestGoRepositoryCompilerTests(t *testing.T) {
 	shards := os.Getenv("CIRCLE_NODE_TOTAL")
 	shard := os.Getenv("CIRCLE_NODE_INDEX")
 	if shards != "" && shard != "" {
-		// We are running under CircleCI parallel test job, so we need to shard execution.
+		// When we are running under CircleCI parallel test job, we need to shard execution.
```

Comment:
Should this be deleted, or is that a follow-up task after Circle CI is finally removed?

Reply:
I was hoping that we could get another pass at CI work sometime to do some cleanup and tryout features like [go-test-split-action](https://github.com/hashicorp-forge/go-test-split-action). I'm sure there are some other things we haven't noticed yet that we'll run across since GHA is fairly new to us.

Reply:
Oooh, another one which I literally ran across this morning about 5 mins ago. During my teams standup I was telling my team about what I was doing. For my [WIP] PR I have a copy of this GHA bumped to go1.20. I made a comment that I only have a few fails left but wasn't sure because CircleCI had been timing out and the GHA was cancelling `GopherJS Test (crypto)` when non-crypto failed. A QA on our team said "just add `fail-fast: false`." After looking at the [fail-fast docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast), they're right, we should probably add that into GopherJS's GHA. Things like that we wouldn't know about that until we were having some failing tests. I'll add it into the WIP PR to try it out as something we can add in a follow up with any other improvements we run across.

Reply:
I am okay with keeping it for now. It doesn't add _that_ much complexity and would make a rollback easier if we have to.

Reply:
@grantnelson-wf actually, mind adding it into this PR too? I've updated repo settings to require GHA checks instead of CircleCI, and I'd like to make sure the CircleCI Pipeline stops showing up after you push something into this PR.

Reply:
Can do!

Reply:
Updated `TestGoRepositoryCompilerTests` to not have the CircleCI flags.
Added in `fail-fast: false` to let all the jobs of GopherJS Tests finish even when one fails.

Reply:
LGTM, thank you!